src/lava/proc/conv/process.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from lava.proc.conv import utils
+
+
+class Conv(AbstractProcess):
+    """Convolution connections between neurons.
+
+    Parameters
+    ----------
+    input_shape : tuple of three ints
+        shape of input to the process in (X, Y, Z) or (W, H, C) format.
+    weight : tensor/array
+        convolution kernel weight. The dimension should be in
+        (C_out, W, H, C_in) format.
+    padding : int or tuple of two ints
+        convolution padding size. Default is 0.
+    stride : int or tuple of two ints
+        convolution stride. Default is 1.
+    dilation : int or tuple of two ints
+        convolution dilation. Default is 1.
+    groups : int
+        number of groups in convolution. Default is 1.
+
+    Note
+    ----
+    padding, stride and dilation are expected in (X, Y) or (W, H) if tuple.
+    """
+    def __init__(self, **kwargs):
+        # the process In/OutPort shapes are considered to be XYZ(WHC) format.
+        # the kernel weight shape is expected to be in (C_out, W, H, C_in)
+        # format.
+        # Why? This is the format that Loihi conv feature uses.
+        super().__init__(**kwargs)
+
+        def broadcast_arg(name, default):
```

Comment:
Broadcasting should be done by the Compiler/ProcModelBuilder when it expands the Var(..., init=...) argument to the Var value at the beginning of execution on the remote process executing a particular ProcModel.
This avoids allocating unnecessary memory at the process level.
Is this broadcasting feature insufficient?

Reply:
It is needed for/simplifies output shape calculation. At this level I can create a var and access var.init but that would still be a scalar. Correct me if I got it wrong.

Reply:
This simplifies output_shape calculation. Var.init at this point would give the scalar value. Correct me if I am wrong.

Reply:
Yes, Var.init will only hold the initially assigned value.
I would have assumed that your output_shape helper function could simply compute the output shape from 'shape'.

Reply:
It's a local function that gets killed after it servers its purpose in init. So I am not too worried about resources here.

Reply:
won't it be good to put this in utils?

Reply:
The reason I am hesitant to put this to utils is in this form, it uses kwargs from init space. This makes these lines simple
`padding = broadcast_arg('padding', 0)`
Whereas it would be
`padding = utils.broadcast_arg(kwargs.get('padding', 0))`
I am happy to do that as well.

Reply:
Yes could go in utils. 

But forget my original comment. I did not read it carefully. I thought you were broadcasting big data arrays here. But you are actually just building a shape tuple. However, in this case the name is 'broadcast' is misleading because you are not actually broadcasting anything.